### PR TITLE
Add failure test for shard move

### PR DESCRIPTION
**Pull request by agedemenli**
DESCRIPTION: Adds failure test for shard move
DESCRIPTION: Remove function `WaitForAllSubscriptionsToBecomeReady` and related tests

Adding some failure tests for shard moves.
Dropping the not-needed-anymore function `WaitForAllSubscriptionsToBecomeReady`, as the subscriptions now start as ready from the beginning because we don't use logical replication table sync workers anymore.

fixes: #6260

### CODE REVIEW

File: src/test/regress/sql/failure_online_move_shard_placement.sql [hunk 1]
```diff
@@ -89,10 +81,6 @@ SELECT master_move_shard_placement(101, 'localhost', :worker_1_port, 'localhost'
 SELECT citus.mitmproxy('conn.onQuery(query="^ALTER SUBSCRIPTION .* DISABLE").cancel(' || :pid || ')');
 SELECT master_move_shard_placement(101, 'localhost', :worker_1_port, 'localhost', :worker_2_proxy_port);
 
--- failure on dropping subscription
-SELECT citus.mitmproxy('conn.onQuery(query="^DROP SUBSCRIPTION").kill()');
-SELECT master_move_shard_placement(101, 'localhost', :worker_1_port, 'localhost', :worker_2_proxy_port);
-
```
Comment (JelteF) [inline src/test/regress/sql/failure_online_move_shard_placement.sql:95]: I think this one was deleted by accident

Comment (agedemenli) [reply src/test/regress/sql/failure_online_move_shard_placement.sql]: oops, reverted, thanks!

File: src/test/regress/sql/failure_split_cleanup.sql [hunk 2]
```diff
@@ -162,99 +160,7 @@ SELECT create_distributed_table('table_to_split', 'id');
...
-    SET search_path TO "citus_failure_split_cleanup_schema", public, pg_catalog;
-    SET citus.show_shards_for_app_name_prefixes = '*';
-    -- Empty child shards after cleanup
-    SELECT relname FROM pg_class where relname LIKE '%table_to_split_%' AND relkind = 'r' order by relname;
-    -- Empty publications
-    SELECT pubname FROM pg_publication;
-    -- Empty replication slot table
-    SELECT slot_name FROM pg_replication_slots;
-    -- Empty subscriptions
-    SELECT subname FROM pg_subscription;
```
Comment (JelteF) [inline src/test/regress/sql/failure_split_cleanup.sql:209]: I think this test should stay

File: src/test/regress/sql/failure_online_move_shard_placement.sql [hunk 3]
```diff
@@ -65,21 +65,13 @@ SELECT master_move_shard_placement(101, 'localhost', :worker_1_port, 'localhost'
...
--- failure on polling last write-ahead log location reported to origin WAL sender
-SELECT citus.mitmproxy('conn.onQuery(query="^SELECT min\(latest_end_lsn").kill()');
+-- failure on dropping subscription
+SELECT citus.mitmproxy('conn.onQuery(query="^ALTER SUBSCRIPTION .* (ENABLE|DISABLE)").kill()');
 SELECT master_move_shard_placement(101, 'localhost', :worker_1_port, 'localhost', :worker_2_proxy_port);
-
--- cancellation on polling last write-ahead log location reported to origin WAL sender
-SELECT citus.mitmproxy('conn.onQuery(query="^SELECT min\(latest_end_lsn").cancel(' || :pid || ')');
```
Comment (JelteF) [inline src/test/regress/sql/failure_online_move_shard_placement.sql:81]: these two tests should stay too

**Review by JelteF — CHANGES_REQUESTED** [review]

**Review by JelteF — APPROVED** [review]

**Pull request merged**